src/libraries/System.Private.CoreLib/src/System/CharEnumerator.cs

```diff
@@ -31,7 +31,7 @@ public bool MoveNext()
             return false;
         }
 
-        public void Dispose() => _str = null!;
+        public void Dispose() => _str = String.Empty;
```

Comment:
```suggestion
        public void Dispose() => _str = string.Empty;
```
We have enforced the rule to only use language keyword (lower case `string`) instead of the type name `String`.